client/dashboard/data/domain-transfer.ts

```diff
@@ -0,0 +1,53 @@
+import wpcom from 'calypso/lib/wp';
+
+export type IpsTag = {
+	tag: string;
+	registrarName: string;
+	registrarUrl: string;
+};
+
+export async function updateDomainLock( domain: string, enabled: boolean ): Promise< void > {
+	return wpcom.req.post( {
+		path: `/domains/${ domain }/transfer`,
+		body: {
+			domainStatus: JSON.stringify( {
+				command: enabled ? 'lock' : 'unlock',
+			} ),
+		},
+	} );
+}
+
+export async function requestTransferCode( domain: string ): Promise< void > {
+	return wpcom.req.post( {
+		path: `/domains/${ domain }/transfer`,
+		body: {
+			domainStatus: JSON.stringify( {
+				command: 'only-send-code',
+			} ),
+		},
+	} );
+}
+
+export async function saveIpsTag( domain: string, ipsTag: string ): Promise< void > {
+	return wpcom.req.post( {
+		path: `/domains/${ domain }/transfer`,
+		body: {
+			domainStatus: JSON.stringify( {
+				command: 'set-ips-tag',
+				payload: { ips_tag: ipsTag },
+			} ),
+		},
+	} );
+}
+
+export async function getIpsTagList(): Promise< IpsTag[] > {
+	try {
+		const response = await fetch( 'https://widgets.wp.com/domains/ips-tag-list.min.json' );
+		if ( ! response.ok ) {
+			throw new Error( `HTTP error! status: ${ response.status }` );
+		}
+		return await response.json();
+	} catch ( error ) {
+		throw new Error( 'Failed to fetch IPS tag list' );
```

Comment:
This is gobbling up the exception that was carefully crafted a couple lines above and looses the status code information.

If you want to wrap the exception you could use a `DashboardDataError` from here:
https://github.com/Automattic/wp-calypso/blob/a2e34f0bfb73603e65a84076c8c2ae379e6b8d29/client/dashboard/data/error.ts#L1

```suggestion
		throw new DashboardDataError( 'ips-tag-list-fail', error );
```

That way we retain the existing error.

Or maybe we don't need to catch at all and we just let the exception and its error message propagate to the regular error handler?